bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/TaskResolutionService.kt

```diff
@@ -86,6 +86,67 @@ class TaskResolutionService(
+                BusinessPartnerSharingError.BpnErrorTooManyOptions,
+                if (task.processingState.errors.isNotEmpty()) task.processingState.errors.joinToString(" // ") { it.description }.take(255) else null
+            )
+            ResultState.MandatoryFieldValidationFailed -> RequestCreationResult.error(
+                sharingState,
+                BusinessPartnerSharingError.MandatoryFieldValidationFailed,
+                if (task.processingState.errors.isNotEmpty()) task.processingState.errors.joinToString(" // ") { it.description }.take(255) else null
+            )
+            ResultState.BlacklistCountryPresent -> RequestCreationResult.error(
+                sharingState,
+                BusinessPartnerSharingError.BlacklistCountryPresent,
+                if (task.processingState.errors.isNotEmpty()) task.processingState.errors.joinToString(" // ") { it.description }.take(255) else null
+            )
+            ResultState.InvalidSpecialCharacters -> RequestCreationResult.error(
+                sharingState,
+                BusinessPartnerSharingError.InvalidSpecialCharacters,
+                if (task.processingState.errors.isNotEmpty()) task.processingState.errors.joinToString(" // ") { it.description }.take(255) else null
+            )
+            ResultState.MandatoryFieldMissing -> RequestCreationResult.error(
+                sharingState,
+                BusinessPartnerSharingError.MandatoryFieldMissing,
+                if (task.processingState.errors.isNotEmpty()) task.processingState.errors.joinToString(" // ") { it.description }.take(255) else null
+            )
+            ResultState.BpnlChanged -> RequestCreationResult.error(
+                sharingState,
+                BusinessPartnerSharingError.BpnlChanged,
+                if (task.processingState.errors.isNotEmpty()) task.processingState.errors.joinToString(" // ") { it.description }.take(255) else null
+            )
+            ResultState.UnclearEntity -> RequestCreationResult.error(
+                sharingState,
+                BusinessPartnerSharingError.UnclearEntity,
+                if (task.processingState.errors.isNotEmpty()) task.processingState.errors.joinToString(" // ") { it.description }.take(255) else null
+            )
+            ResultState.UnknownSpecialCharacters -> RequestCreationResult.error(
+                sharingState,
+                BusinessPartnerSharingError.UnknownSpecialCharacters,
+                if (task.processingState.errors.isNotEmpty()) task.processingState.errors.joinToString(" // ") { it.description }.take(255) else null
+            )
+            ResultState.MatchBasedOnProvidedNameOrIdentifier -> RequestCreationResult.error(
+                sharingState,
+                BusinessPartnerSharingError.MatchBasedOnProvidedNameOrIdentifier,
+                if (task.processingState.errors.isNotEmpty()) task.processingState.errors.joinToString(" // ") { it.description }.take(255) else null
```

Comment:
This looks unnecessarily verbose. Currently, the Gate's sharing state only has up to one error it can display. I would propose to use the generic SharingProcessError type if there are several errors, just like it is currently done. And if there is only one error we can save the more specific error here.

For this I would then use a when-clause mapping from Task error type to Gate error and then map to a RequestCreationResult afterwards.

Proposal somethind like this

```kotlin
fun handleError(...) : RequestCreatinResult{
 if(errors.size == 1){
   when(taskErrorType){
       UnkownSpecialCharacters -> GateErrorType.UnknownSpecialCharacters
       ...
   }.map{ gatErrorType -> RequestCreationResult.erro(...)   }
 }else{
  RequestCreationResult.error(...) like before
 }
}
```